### ISSUE TITLE
about set QInteractorStyleVesselSegmentation save to vtkPolygon

### ISSUE THREAD

**Issue by wuzhuobin** [wuzhuobin/blood_vessel#36]
save contour to close loop vtkPolygon instead of vtkPolyLine can have better performance in the fill polygon part

**Comment by alabamagan** [comment]
Hmm why is that? How did you find that out?

Get Outlook for iOS




On Fri, Dec 2, 2016 at 11:37 PM +0800, "wuzhuobin" <notifications@github.com> wrote:












save contour to close loop vtkPolygon instead of vtkPolyLine can have better performance in the fill polygon part

**Comment by wuzhuobin** [comment]
anyway, even though 
polygon cell -> vtkVoxelContoursToSurfaceFilter -> vtkPolyDataToImageStencil -> vtkImageStencil -> overlay Image
above pipeline works fine( because i mix up with the vtkImage Origin and spacing between polydata and vtkImageData).

But it didn't works when there is one slice whose neighbor slice has not contour. it only works in the situation that all contours are successive.

**Comment by wuzhuobin** [comment]
decide to use old method to fill with animation. wtf. 
or using different interpolation method to generate spline like polygon to do filling slice by slice.
but the original interpolation method is vtkBezierContourLineInterpolator, which is only used in contour

**Comment by wuzhuobin** [comment]
dont use 				//spline->ClosedOn();l
this is a trick, which make the first point(0) and the last point merge
if a polygon which has a same point with 2 point id 
if polygon->findPointsInPolygon will get wrong

**Comment by alabamagan** [comment]
Yes, that's why my polygon draw check if there are overlapping points before using it for InPolygon check. 

You can reference to 
_FillPolygon()_
function

**Comment by wuzhuobin** [comment]
anyway, now i using the vtkParametricSpline function for interpolating contour point, although the splines are not perfectly the same as the contour line which interpolate by vtkBezierContourLineInterpolator, but it is supposed enough for coarse filling. And it is extremely fast. 55 slices take about a few seconds in release.

**Comment by alabamagan** [comment]
WOW, that's pretty amazing improvements!

**Comment by wuzhuobin** [comment]
![image](https://cloud.githubusercontent.com/assets/17057846/20999232/2a208d98-bd4e-11e6-9d5b-e34a16be01fe.png)
it might be wrong in this situation... 

future improvement: vtkBezierSpline.  maybe do it one day

**Comment by alabamagan** [comment]
lol, leave it to user la
